static/js/jira-server-url.js

```diff
@@ -55,14 +58,51 @@ const setErrorMessage = error => {
 	$("#gheServerURL").addClass("has-error");
 };
 
-/**
- * Hides the error messages
- */
 const hideErrorMessage = () => {
 	$("#gheServerURLError").hide();
 	$("#gheServerURL").removeClass("has-error");
 };
 
+const activeRequest = () => {
+	$("#gheServerBtnText").hide();
+	$("#gheServerBtnSpinner").show();
+};
+
+const requestFailed = () => {
+	$("#gheServerBtnText").show();
+	$("#gheServerBtnSpinner").hide();
+};
+
+const verifyGitHubServerUrl = (gheServerURL, installationId) => {
+	const csrf = document.getElementById('_csrf').value;
+
+	window.AP.context.getToken(function(token) {
+		$.post("/jira/server-url", {
+				gheServerURL,
+				_csrf: csrf,
+				jwt: token,
+				jiraHost,
+				installationId
+			},
+			function(data) {
+				if (data.success) {
+					const pagePath = data.moduleKey;
+					AP.navigator.go(
+						"addonmodule",
+						{
+							moduleKey: pagePath
+						}
+					);
+				} else {
+					console.error(`Failed to verify GHE server url. ${gheServerURL}`);
+					requestFailed();
+					// TODO - render error component
```

Comment:
Going to add the partial as part of [ARC 1473](https://softwareteams.atlassian.net/jira/software/projects/ARC/boards/385?assignee=604894ffe19f910069497c6a&selectedIssue=ARC-1473)